src/cmps/tags/Tags.tsx

```diff
@@ -1,51 +1,92 @@
-import React, { Component } from "react";
+import React, { Component } from 'react';
 import './Tags.css';
 import Tag from '../tag/Tag';
-// eslint-disable-next-line
-import { string } from "prop-types";
+// import { string, number } from 'prop-types';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { library } from '@fortawesome/fontawesome-svg-core';
+import { faPlus } from '@fortawesome/free-solid-svg-icons';
+// import { any, element } from 'prop-types';
 
+library.add(faPlus);
 
 class Tags extends Component {
-
     state = {
-        tags: [{
-            id: 1,
-            name: 'tag1'
-        }, {
-            id: 2,
-            name: 'tag2'
-        }]
-    }
+        currentTag: '',
+        tags: [],
+    };
 
-    remove = () => {
-        console.log('button X clicked');
-    }
+    remove = (id: string) => {
+        let index = this.state.tags.findIndex((element: any) => element.id === id);
+        this.setState(this.state.tags.splice(index, 1));
```

Comment:
never use current state in `setState`.

you can find why [here](https://www.freecodecamp.org/news/get-pro-with-react-setstate-in-10-minutes-d38251d1c781/) - search for the title: State Updates May Be Asynchronous

ask me if it is not clear!

Reply:
I did not make it...